lib/internal/modules/run_main.js

```diff
@@ -131,19 +134,36 @@ async function handleMainPromise(promise) {
  * Because of backwards compatibility, this function is exposed publicly via `import { runMain } from 'node:module'`.
  * @param {string} main - First positional CLI argument, such as `'entry.js'` from `node entry.js`
  */
+let isCommandMain = false;
 function executeUserEntryPoint(main = process.argv[1]) {
-  const resolvedMain = resolveMainPath(main);
-  const useESMLoader = shouldUseESMLoader(resolvedMain);
+  const mainPath = resolveMainPath(main);
+  const useESMLoader = shouldUseESMLoader(mainPath);
   if (useESMLoader) {
-    runMainESM(resolvedMain || main);
+    runMainESM(mainPath || main, isCommandMain);
   } else {
     // Module._load is the monkey-patchable CJS module loader.
     const { Module } = require('internal/modules/cjs/loader');
     Module._load(main, null, true);
   }
+  isCommandMain = false;
+}
+
+/*
+ * This is a special function that can be called before executeUserEntryPoint
+ * to note that the coming entry point call is a command main.
+ *
+ * This should really just be implemented as a parameter, but executeUserEntryPoint is
+ * exposed publicly as `runMain` which has backwards-compatibility requirements, hence
+ * this approach.
```

Comment:
Would it not be backward-compatible to just add an extra parameter at the end?

Reply:
I've updated the comment to note this is not actually a backwards compat concern so much as a concern that we mustn't expose setting the command main to userland as it is an entirely internally-defined concept.

Reply:
Surely there’s a less awkward workaround than calling a function that sets a module-scope property ahead of time? Since `runMain` gets called multiple times, like for `--import` and so on, this approach feels error-prone. Perhaps the additional parameter could be an undocumented options object, and Node core checks that this object contains a particular symbol on it? And this symbol would be something that’s defined internally and therefore unusable by user code.